src/index.ts

```diff
@@ -32,16 +33,24 @@ interface UserInput {
   birthDate: string;
 }
 
+const prisma = new PrismaClient();
+
 const resolvers = {
   Query: {
     hello: () => 'Hello World',
   },
   Mutation: {
-    createUser: (_, args: { data: UserInput }) => {
-      return {
-        id: 1,
-        ...args.data,
-      };
+    createUser: async (_, args: { data: UserInput }) => {
+      const userData = args.data;
+
+      return await prisma.user.create({
+        data: {
```

Comment:
se usar 1, aqui vc pode omitir o objeto, dá uma bela encurtada
```suggestion
        data: { name, email, password, birthDate } 
```
2 agora o 2º, como o objeto userData é o mesmo de data, vc pode simplesmente
```suggestion
        data: userData
```
hahaha

Reply:
Resolvido pelo caminho 1